### DISCUSSION TITLE
Filter Assets by Tags with AssetQueryBuilder with wherehas

### DISCUSSION THREAD

**Discussion by nicorttngr** [statamic/cms#7520] [Q&A] [answered]
I'm creating a paginated grid of all assets in a container with a filter function. The assets are tagged and i'm trying to build a query which checks for the filtered tags. 
At first I was trying to do this:

```php
$arrayOfFilteredTerms = ['tags::foo', 'tags::bar'];

Asset::query()->where('container', 'media')->whereHas('tags', function($q) use ($arrayOfFilteredTerms){
        $q->whereIn('id', $arrayOfFilteredTerms);
});
```
which results in 

`Call to undefined method Statamic\Assets\QueryBuilder::whereHas()`

When the AssetQueryBuilder doesn't work with "whereHas",  I have to filter my collection **after** getting the query. This collides with the pagination - which limits the query results before they get filtered - and besides it's very slow.

Is there any chance to filter assets by tags directly in the query?

**Comment by jasonvarga** [comment, marked as answer]
We don't support `whereHas` at the moment, as you've noted.

But you can just treat it as json.

```php
Asset::query()->where('container', 'media')->whereJsonContains('tags', $arrayOfFilteredTerms)
```

**Reply by nicorttngr** [reply]
worked like a charm! Just had to remove "tags::" from the term string. thank you!
